HSEARCH-3418 Document that Instants are truncated to the millisecond when indexed

yrodiere opened the pull request:
https://hibernate.atlassian.net/browse/HSEARCH-3418

fax4ever commented on documentation/src/main/asciidoc/mapper-orm-mapping.asciidoc:
```diff
@@ -479,47 +479,108 @@ see <<backend-lucene-field-types,Lucene field types>>
+|java.time.MonthDay|-|-|`MonthDay.parse(String)`
+|java.util.UUID|`toString()` as a `java.lang.String`|-|`UUID.fromString(String)`
+|java.util.Calendar
+ |A `java.time.ZonedDateTime` representing the same date/time and timezone; see <<mapper-orm-legacy-date-time-apis>>.
```
Maybe we can put the `<<mapper-orm-legacy-date-time-apis>>`s under the limitations column.

yrodiere commented:
Yes, the limitations are _currently_ the same. I am not sure it will stay that way, though, so I'd rather document them separately. It's not like many people will need to use both backends anyway ;)

I addressed your comment and rebased. Will merge as soon as CI goes green.
